src/ORM/DataObject.php

```diff
@@ -369,6 +369,15 @@ class DataObject extends ModelData implements DataObjectInterface, i18nEntityPro
      */
     private static ?DataObjectSchema $schema = null;
 
+    /**
+     * A list of DataObject subclasses whose query caches should be reset whenever resetting
+     * the query cache for this class.
+     *
+     * Do not include other classes from the current hierarchy in this list.
+     * i.e. Page should not include SiteTree or vice versa.
+     */
+    private static array $query_cache_dependent_classes = [];
```

Comment:
This allows us to be 1000000% sure that cached permission queries don't cause problems when e.g. updating the groups for a member